py-polars/polars/functions/as_datatype.py

```diff
@@ -502,6 +502,30 @@ def concat_list(exprs: IntoExpr | Iterable[IntoExpr], *more_exprs: IntoExpr) ->
     return wrap_expr(plr.concat_list(exprs))
 
 
+def array(
```

Comment:
Can we make this API be simply array(vals: Iterable[IntoExpr], *, datatype:...)? Then it is the same semantics of pythons builtin list(), and it is unambiguous as to whether or not we remove a level of nesting if we pass in multiple list expressions (we don't).

Reply:
Thanks @NickCrews, I think I agree with you on this point but I'm a bit fuzzy on the endpoint. That is, I think it would help to exemplify to understand exactly what syntax would be allowed. Let's say we have a signature of `array(vals: Iterable[IntoExpr], datatype:...)` - here I'm dropping the second argument of `*` since I think we want just the one named argument.
So we can have expressions like
`pl.array(['a', 'b', 'c'])` for a set of columns
and expressions like `pl.array([f"A_lag_{i}" for i in range(3)][::-1])` (from the concat_list docs for a rolling window).

But where I start to get fuzzy is things like column globs. I think it is valuable to have syntax like
`pl.array(pl.all())`
The `all` function just returns a single expression (https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.all.html#polars.all) so this may not be compatible with the idea of taking an iterable.
Anyway, I think there are a few permutations here and I think it would be helpful to have some concrete examples and how they interact with the expression syntax.